### ISSUE TITLE
ftm example hangs with... wifi:Starting FTM session with ... in 4227.910 Sec (IDFGH-6072)

### ISSUE THREAD

**Issue by crackwitz** [espressif/esp-idf#7753]
## Environment

- Development Kit:      ESP32-S2-DevKitM
- Kit version: v1?
- Module or chip used:  ESP32-S2-Mini-1
- IDF version (run ``git describe --tags`` to find it): v4.3.1
- Build System:         idf.py (whatever's standard when using the vscode extension)
- Operating System:     Windows
- (Windows only) environment type: not sure, using vscode extension
- Using an IDE?: vscode
- Power Supply:         USB

## Problem Description

Running `ftm` example, repeatedly requesting FTM as `ftm -s esp-ftm -c 16`, eventually one session reports a ridiculous amount of time for the "request" and then the ESP32 is just stuck there:

> I (70092) ftm_station: Requesting FTM session with 16 frames, burst 200 ms (0: No Pref)
> W (70102) wifi:Starting FTM session with ... in **4227.910 Sec**
> W (70102) wifi:Mode: non-ASAP, Bursts: 2, FTM's per burst: 8, Burst Period: 200mSec, Burst Duration: 32000uSec

I'm guessing some timeout calculation wrapped wrong and it's waiting for the wrong timestamp or something.

I searched for (parts of) that (format) string but couldn't find it in the source, so it must be in the closed-source parts (radio blob). I found an instance of it in `libnet80211.a` for esp32-s2.

### Expected Behavior

It should report a time of less than one second, as do all other invocations of the ftm command.

### Actual Behavior

Reports an amount of time it definitely did not take, and then gets stuck.

### Steps to reproduce

1. run two ESP32-S2 with ftm example
2. send one to AP mode, let's give it the SSID `esp-ftm`
3. repeatedly send `ftm -s esp-ftm -c 16` or something similar
4. eventually, it's hard to reproduce, it will act as described above

### Code to reproduce this issue

No code, just `ftm` example.

## Debug Logs

Crucial parts, see above. Startup log here:
```
ESP-ROM:esp32s2-rc4-20191025
Build:Oct 25 2019
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3ffe6100,len:0x1840
load:0x4004c000,len:0xa04
load:0x40050000,len:0x2b80
SHA-256 comparison failed:
Calculated: 375bd72a9749f44bc2ff91bd365cd3ebe0c3bdc7f5f093c6a79ed69276b80aa2
Expected: 2aa35969f13614ed6ba0f2848e29c0ce26b6a59ee8bc623b30b9f0664fcf07b4
Attempting to boot anyway...
entry 0x4004c1f4
I (40) boot: ESP-IDF v4.3.1-dirty 2nd stage bootloader
I (40) boot: compile time 15:11:42
I (40) boot: chip revision: 0
I (43) boot.esp32s2: SPI Speed      : 80MHz
I (48) boot.esp32s2: SPI Mode       : DIO
I (52) boot.esp32s2: SPI Flash Size : 4MB
I (57) boot: Enabling RNG early entropy source...
I (62) boot: Partition Table:
I (66) boot: ## Label            Usage          Type ST Offset   Length
I (73) boot:  0 nvs              WiFi data        01 02 00009000 00006000
I (81) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (88) boot:  2 factory          factory app      00 00 00010000 00100000
I (96) boot: End of partition table
I (100) esp_image: segment 0: paddr=00010020 vaddr=3f000020 size=17b54h ( 97108) map
I (128) esp_image: segment 1: paddr=00027b7c vaddr=3ffcc410 size=03d58h ( 15704) load
I (132) esp_image: segment 2: paddr=0002b8dc vaddr=40022000 size=0473ch ( 18236) load
I (138) esp_image: segment 3: paddr=00030020 vaddr=40080020 size=86730h (550704) map
I (252) esp_image: segment 4: paddr=000b6758 vaddr=4002673c size=15cd4h ( 89300) load
I (274) esp_image: segment 5: paddr=000cc434 vaddr=40070000 size=00054h (    84) load
I (274) esp_image: segment 6: paddr=000cc490 vaddr=50000000 size=00010h (    16) load
I (293) boot: Loaded app from partition at offset 0x10000
I (293) boot: Disabling RNG early entropy source...
I (305) cach▒~▒▒▒~Սѥ▒▒▒cache    : size 8KB, 4Ways, cache line size 32Byte
I (305) cpu_start: Pro cpu up.
I (319) cpu_start: Pro cpu start user code
I (319) cpu_start: cpu freq: 160000000
I (319) cpu_start: Application information:
I (324) cpu_start: Project name:     ftm
I (328) cpu_start: App version:      4b39a16-dirty
I (334) cpu_start: Compile time:     Oct 22 2021 15:11:33
I (340) cpu_start: ELF file SHA256:  c10e0d81f677fca9...
I (346) cpu_start: ESP-IDF:          v4.3.1-dirty
I (351) heap_init: Initializing. RAM available for dynamic allocation:
I (358) heap_init: At 3FF9E054 len 00001FAC (7 KiB): RTCRAM
I (365) heap_init: At 3FFD49C0 len 00027640 (157 KiB): DRAM
I (371) heap_init: At 3FFFC000 len 00003A10 (14 KiB): DRAM
I (378) spi_flash: detected chip: generic
I (382) spi_flash: flash io: dio
I (390) cpu_start: Starting scheduler on PRO CPU.
I (422) system_api: Base MAC address is not set
I (422) system_api: read default base MAC address from EFUSE
I (422) wifi_init: rx ba win: 6
I (432) wifi_init: tcpip mbox: 32
I (432) wifi_init: udp mbox: 6
I (432) wifi_init: tcp mbox: 6
I (432) wifi_init: tcp tx win: 5744
I (442) wifi_init: tcp rx win: 5744
I (442) wifi_init: tcp mss: 1440
I (442) wifi_init: WiFi IRAM OP enabled
I (452) wifi_init: WiFi RX IRAM OP enabled
I (452) phy_init: phy_version 1800,e7ef680,Apr 13 2021,11:45:08

Type 'help' to get the list of commands.
Use UP/DOWN arrows to navigate through command history.
Press TAB when typing command name to auto-complete.

 ==========================================================
 |                      Steps to test FTM                 |
 |                                                        |
 |  1. Use 'help' for detailed information on parameters  |
 |  2. Start SoftAP with command 'ap <SSID> <password>'   |
 |                          OR                            |
 |  2. Use 'scan' command to search for external AP's     |
 |  3. On second device initiate FTM with an AP using     |
 |     command 'ftm -I -s <SSID>'                         |
 ==========================================================

ftm>
```

**Comment by crackwitz** [comment]
oh by the way, I believe the format string for `"Starting FTM session with"` uses just `%d` for the milliseconds, when it should use `%03d`... I get funny decimals sometimes that make more sense if interpreted as milliseconds (`97` = `0.097`) instead of fractions of a second (`0.97`).

**Comment by crackwitz** [comment]
Reporting back...

```
W (2273395) wifi:Start time delay for FTM too high! (4227.982 Sec)
```
I just got this running from git master of today.

So I'm glad it's now being caught and handled well (!) from inside the wifi blob, but you are probably interested in knowing it can still happen :)

whole section (wifi ftm example plus some of my own changes: hammering with FTM sessions, forcing a rescan on failure):
```
I (2272995) fixed: FTM session with 7c:df:a1:40:cd:a5, 16 frames, burst 200 ms
W (2272995) wifi:Starting FTM session with 7c:df:a1:40:cd:a5 in 0.113 Sec
W (2272995) wifi:Mode: non-ASAP, Bursts: 2, FTM's per burst: 8, Burst Period: 200mSec, Burst Duration: 32000uSec
W (2273335) wifi:FTM session ends with 15 valid readings out of 15, Avg raw RTT: 0.0 nSec, Avg RSSI: -32
I (2273335) fixed: RTT result: ToF 0.000 ns, 0.000 m
I (2273395) fixed: FTM session with 7c:df:a1:40:cd:a5, 16 frames, burst 200 ms
W (2273395) wifi:Start time delay for FTM too high! (4227.982 Sec)
I (2273395) ftm_station: FTM with 7c:df:a1:40:cd:a5 failed, status 4
W (2273405) fixed: FTM: failure, status 4
I (2273455) ftm_station: Scanning for esp32 FTMR
I (2275955) ftm_station: sta scan done
I (2275955) fixed: FTM session with 7c:df:a1:40:cd:a5, 16 frames, burst 200 ms
W (2275955) wifi:Starting FTM session with 7c:df:a1:40:cd:a5 in 0.151 Sec
W (2275965) wifi:Mode: non-ASAP, Bursts: 2, FTM's per burst: 8, Burst Period: 200mSec, Burst Duration: 32000uSec
W (2276335) wifi:FTM session ends with 15 valid readings out of 15, Avg raw RTT: 0.0 nSec, Avg RSSI: -33
I (2276345) fixed: RTT result: ToF 0.000 ns, 0.000 m
```
